src/base/bittorrent/tracker.h

```diff
@@ -49,21 +50,76 @@ namespace Http
 
 namespace BitTorrent
 {
-    struct Peer
+    class Peer
     {
-        QString ip;
-        QString peerId;
-        int port;
+    public:
+        Peer() = default;
+        Peer(QString&& ip, QString&& announce_ip, std::string&& peer_id, int port, QByteArray&& uid);
+
+        Peer(const Peer&) = default;
+        Peer(Peer&&) = default;
+        Peer& operator=(const Peer&) = default;
+        Peer& operator=(Peer&&) = default;
+
+        const QString& getIp() const;
 
         bool operator!=(const Peer &other) const;
         bool operator==(const Peer &other) const;
-        QString uid() const;
-        libtorrent::entry toEntry(bool noPeerId) const;
+        const QByteArray& uid() const;
+        // converts to appropriate entry taking into account whether 'clientIp' is a public ip address,  i.e. provides
+        // public ip if 'clientIP' belongs to a public network, otherwise gives connection ip
+        libtorrent::entry toEntryForClient(bool noPeerId, const QString& clientIp) const;
```

Comment:
Since you use PeerBuilder to build a peer no need to implement Peer as class at all. Simple struct is enough here.
As for this method (toEntryForClient()), it has private meaning and can be simply transformed to module wide function `peerToEntry()`. The main idea is to move all private things to .cpp.